### DISCUSSION TITLE
Error when fitting the TweedieGLM

### DISCUSSION THREAD

**Discussion by karlzw** [casact/chainladder-python#421] [Q&A] [answered]
I am trying to calculate my reserve using the gamma distribution found in the TweedieGLM class.
However, I keep getting the following error: 

ValueError: Some value(s) of y are out of the valid range for family TweedieDistribution

What am I doing wrong?

**Comment by jbogaardt** [comment, marked as answer]
Hard to say without a reproducible example, but the gamma error structure requires that the incremental values of your triangle be non-negative, so if you have negative values (usually associated with a reduction in case reserves or a recovery), then the data doesn't fit the model distribution.

**Reply by karlzw** [reply]
Here's the cumulative triangle (quarterly) I am using. 

[triangle_2017.xlsx](https://github.com/casact/chainladder-python/files/11006801/triangle_2017.xlsx)

**Reply by jbogaardt** [reply]
Thanks.  Correction to my previous comment, the Gamma error is strictly non-negative.  0 valued incrementals are inadmissible for gamma, but your triangle has some causing the estimator to fail.

**Reply by karlzw** [reply]
Thanks, that solved my issue
